content/knowledge_base/glossary.md

```diff
@@ -2,11 +2,29 @@
 
 ## General Blockchain Concepts
 
-### Consensus
+### Blockchain
 
-TODO: add
+This technology is based on a chain of blocks linked to each other.
+Once a new block is created and inserted on the chain, the new block will have a link to the previous block, creating a chain.
+They all include the hash of the previous block except for the first one, which is called the genesis block and has a zero hash value.
 
-### Blockchain
+There are two types of blockchain &#0151 permissionless and permissioned.
+The permissionless blockchain, also called public blockchain, is available to everyone, and, because of that, anyone can create and track down blocks.
+This type of blockchain comes with some limitations, mainly slow performance and bad scaling.
+
+On the other side, the permissioned blockchain, or often referred to as private blockchain, has limited availability, only being accessible to a certain group instead of being publicly available, they are also faster and better to scale.
+On this type of blockchain, there is no anonymity like in a public blockchain.
+
+The core concepts of a blockchain network are:
+
+* Transparency
+    * Every full node in the network has a copy of the whole chain of blocks. This means that every transaction is available to each member, making the transactions traceable. This makes every transaction more transparent and removes the need for intermediaries, allowing the process to be more efficient.
+* Immutable
+    * Information in the blockchain is immutable. If a block in the middle of the chain is changed, the chain is no longer valid. To be valid, the blocks in the furthen down the chain must be recomputed. To apply the previous changes in every node in the network, a consensus algorithm should be used in every block that was modified.
+* Decentralization
+    * This is one of the biggest advantages of the blockchain, being a decentralized system allows for the lack of a central authority to control the transactions. Every full node has a copy of the chain, which they can update with new information.  If someone wants to update the chain, they need to have the consensus of the network. This is done with a consensus algorithm.
```

Comment:
```suggestion
```

Reply:
Please do not accept as 100% true something about blockchain on the internet that can just be copied

Most of the articles today are written just for comparison and getting more clicks and come from people that have no idea about it and write utter nonsense.

Reply:
What do you mean? This was written by me, sourcing different papers.

Reply:
Yes, but it depends from where you get the info.

If you read what I added as a description you will see that:

1.  There are more types of blockchains than permissionless and permissioned - there are also public and private.
Permissionless and permissioned types differ based on permission granted to access the network, being with public or private data. Some peers from these networks might have access to all of the data and some not.
Example: https://www.r3.com/blog/how-public-permissioned-blockchains-are-not-an-oxymoron-2/

2. Public and private blockchains separate based on data different groups of peers have access to. They can be permissionless or permissioned themselves - based on if there is an access level for the peers or not.

3. These types of networks usually lack most of the `so great` properties of the blockchains, that everybody is telling about - they do not have 100% distribution or decentralization, because:
-  they have different groups of peers doing something different things than other groups of peers - for example a messaging system that can be implemented in the application - although it might also be distributed - if its peers stop working or get malicious the messaging of the whole system will get corrupted => this is no full distribution and decentralization;
- they have different levels of access of data - some peers that are members of organizations inside the networks have access to whole ledger history and the peers that are not parts of these organizations can access some limited amount of blocks in history => there is no full access to the ledger and there is no full trasparency; not to mention that in private networks very often separate organizations of peers follow their own ledgers and its a custom solution who knows what
- these networks do not provide any kind of anonymity or some level of it, meaning they operate in presence of some trust, and `that is why` they do not need all of the crypto-`things` to do to every transaction (and also gain performance from this)

Concluding that transparency, decentralization and distribution are so good characteristics of blockchains is not true for very large part of the networks described:
Transparency: you dont have transparency of ledger data in all peers
Decentralization: you do not have decentralization because of the segmentation in the above types
The only thing that you have in all of these blockchains is immutability, although I will not be surprised if they found a way to amend this in some AI application or some solution with quantum cryptography

The description says that permissionless, fully distributed networks are much slower and not good to scale as the permissioned - but `they` are which actually have the characteristics described in the end

4. Blockchain is a `ledger`. Not just a chain of blocks - it is a chain of blocks from the outside but the idea of the blocks is to group transactions. And the transactions are what updates the ledger state.
The idea behind the distributed ledger is to update it and do the best to tell every participant of the network about its new correct value. This is similar problem to the CAT problem in distributed databases.  This is what all the consensus algorithms are all about roughly speaking.

5. I have added that the blockchain transaction are not only money - you can gain tokens for some online event or a game by paying 15 times with your visa in a certain store this month - thats an actual blockchain that is used in China today; or get some tokens for your social record.
6. In the blockchain (and similarly in the distributed networks), when you gain something you pay for it:
- you gain speed but give away trust, permissionlessnes and data content transparency - distribution and decentralization
- you gain 100% anonymity but give speed and better scaling
- you gain 100% permissionless for arriving and leaving peers but pay with speed and performance

`Real` decentralization is hard and pricey to get.
Also, have in mind that the permissioned, private and in general not 100% decentralized network blockchains are usually just a chain of states changes for one or more ledgers from the network perspective - because a peer in one organization just updates a value in its own ledger and let anyone else about it - like it is with Hyperledger Fabric networks. With networks that are p2p you have a `tree` from the network perspective because some parts of the peers are following branches of the main chain; they eventually understand which is the correct chain, but at any moment of time - there are groups that follow some branched chain and know about different chain of transactions; you do not have 100% of the peers following the same chain at the same time - another price for good decentralization. 

Blockchains are a modern field and more and more people try go gain from it - some % of the information in the net as I said is nonsense, and you dont know what % - what I said was no to take it for 100% true .
If you read what I wrote in the description you will find the dirrefences